extension/httpfs/httpfs.cpp

```diff
@@ -327,47 +383,66 @@ void HTTPFileSystem::Read(FileHandle &handle, void *buffer, int64_t nr_bytes, id
 		hfh.file_offset = location;
 	}
 
-	while (to_read > 0) {
-		auto buffer_read_len = MinValue<idx_t>(hfh.buffer_available, to_read);
-		if (buffer_read_len > 0) {
-			D_ASSERT(hfh.buffer_start + hfh.buffer_idx + buffer_read_len <= hfh.buffer_end);
-			memcpy((char *)buffer + buffer_offset, hfh.read_buffer.get() + hfh.buffer_idx, buffer_read_len);
-
-			buffer_offset += buffer_read_len;
-			to_read -= buffer_read_len;
-
-			hfh.buffer_idx += buffer_read_len;
-			hfh.buffer_available -= buffer_read_len;
-			hfh.file_offset += buffer_read_len;
-		}
-
-		if (to_read > 0 && hfh.buffer_available == 0) {
-			auto new_buffer_available = MinValue<idx_t>(hfh.READ_BUFFER_LEN, hfh.length - hfh.file_offset);
+	if (!hfh.range_read) {
+		GetRequest(hfh, hfh.path, {});
```

Comment:
I'm pretty sure this does a Get request for every `HTTPFileSystem::Read(FileHandle &handle, void *buffer, int64_t nr_bytes, idx_t location)`, a function that is not just internally used in the `HTTPFileSystem::Read(FileHandle &handle, void *buffer, int64_t nr_bytes)`, but part of the Filesystem API. 

As discussed with @pdet already: we need some test to ensure we make the correct amount of request (1 get request for the whole file) This can be done with a regex test on the explain analyze output of a query, since that prints the number of requests that have been made. To force the single file download for a parquet file, we probably want to add either a `force_full_file_download` or add this to the `pragma enable_verification` somehow.

Reply:
I think I'm extending the `pragma enable_verification` to run every query with the full download statement (besides using the range queries).

Reply:
nvm the pragma beauty doesn't work here, going for an option